ci/date-check/src/main.rs

```diff
@@ -35,50 +36,56 @@ impl fmt::Display for Date {
     }
 }
 
-fn make_date_regex() -> Regex {
-    Regex::new(
-        r"(?x) # insignificant whitespace mode
-        <!--\s*
-        [dD]ate:\s*
-        (?P<y>\d{4}) # year
-        -
-        (?P<m>\d{2}) # month
-        \s*-->",
-    )
-    .unwrap()
+fn make_date_regex() -> Vec<Regex> {
+    let patterns = [
+        r"<!--\s+date-check:\s+(\w+)\s+(\d+{4})\s+-->",
+        r"<!--\s+date-check\s+-->\s+(\w+)\s+(\d+{4})",
+    ];
+    let set = RegexSet::new(&patterns).unwrap();
+    set.patterns()
+        .iter()
+        .map(|pattern| Regex::new(pattern).unwrap())
+        .collect()
```

Comment:
What does this usage of `RegexSet` do? It seems like has the same result as just returning `vec![Regex::new(r1), Regex::new(r2)]`.

Reply:
a confused mess is what it is